### PR TITLE
[MRESOLVER-63] FileTransformer#transformData(File) should also throw TransformException

### PR DESCRIPTION
**Pull request by mcimbora**
Up-for-grabs #DevoxxBE

### CODE REVIEW

File: maven-resolver-api/src/main/java/org/eclipse/aether/transform/FileTransformationException.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+import java.io.File;
+
+import org.eclipse.aether.RepositoryException;
+
+/**
+ * Thrown when file transformation failed.
+ */
+public class FileTransformationException
+        extends RepositoryException
```
Comment (rfscholte) [inline maven-resolver-api/src/main/java/org/eclipse/aether/transform/FileTransformationException.java]: This is not really a repository exception. Consider the following usecase: we want to transform the pom.xml during upload with xslt. When trying to generate the XMLParser you must catch the SaxException. This one should be wrapped and thrown by this method.

File: maven-resolver-api/src/main/java/org/eclipse/aether/transform/FileTransformationException.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+{
+
+    private final transient File file;
+
+    /**
+     * Creates a new exception with the specified file.
+     *
+     * @param file The file which transformation failed, may be {@code null}.
+     */
+    public FileTransformationException( File file )
```
Comment (rfscholte) [inline maven-resolver-api/src/main/java/org/eclipse/aether/transform/FileTransformationException.java]: There's not always a file involved yet, no need to use this.

File: maven-resolver-api/src/main/java/org/eclipse/aether/transform/FileTransformer.java [hunk 3]
```diff
@@ -47,5 +46,5 @@
      * @param file the file with the original data
      * @return the transformed data
      */
-    InputStream transformData( File file ) throws IOException;
+    InputStream transformData( File file ) throws FileTransformationException;
```
Comment (rfscholte) [inline maven-resolver-api/src/main/java/org/eclipse/aether/transform/FileTransformer.java]: Both IOException and TransformException must be thrown. Better not make it too explicit, we might do other transformations as well in the future, not only filebased.

File: maven-resolver-api/src/main/java/org/eclipse/aether/transform/FileTransformationException.java (hunk 2 above)
Comment (mcimbora) [reply maven-resolver-api/src/main/java/org/eclipse/aether/transform/FileTransformationException.java]: Removed.

File: maven-resolver-api/src/main/java/org/eclipse/aether/transform/FileTransformer.java (hunk 3 above)
Comment (mcimbora) [reply maven-resolver-api/src/main/java/org/eclipse/aether/transform/FileTransformer.java]: Included TransformException in the throws clause.

File: maven-resolver-api/src/main/java/org/eclipse/aether/transform/FileTransformationException.java (hunk 1 above)
Comment (mcimbora) [reply maven-resolver-api/src/main/java/org/eclipse/aether/transform/FileTransformationException.java]: Good point, thanks for the explaination and prompt review. The exception now inherits directly from Exception.

**Comment by rfscholte** [comment]
If you want you can add yourself as a contributor in the pomfile as well.

**Comment by mcimbora** [comment]
@rfscholte PR updated, added contributor section to pom.xml.

**Comment by jira-importer** [comment]
Resolve #844

**Pull request merged**